DBMS-Assignment-2/Question1.sql

```diff
@@ -0,0 +1,131 @@
+CREATE DATABASE StoreFront;
+USE StoreFront;
+#Question 1
+#Create all tables of eCommerce Application: StoreFront 
+#TABLE users
+CREATE TABLE users(
+user_id INT PRIMARY KEY AUTO_INCREMENT ,
+first_name VARCHAR(25) NOT NULL ,
+last_name VARCHAR(25) ,
+email_id VARCHAR(50) NOT NULL,
+user_password VARCHAR(10) NOT NULL,
+user_type VARCHAR(7) CHECK (user_type IN('admin','shopper'))
+);
+
+#TABLE address
+CREATE TABLE address(
+address_id INT PRIMARY KEY AUTO_INCREMENT,
+user_id INT ,
+area VARCHAR(25) NOT NULL,
+city VARCHAR(25) NOT NULL ,
+state VARCHAR(25) NOT NULL ,
+country VARCHAR(25) DEFAULT "india",
+pin_code INT NOT NULL,
+FOREIGN KEY (user_id) REFERENCES users(user_id)
+);
+
+#TABLE category
+CREATE TABLE category(
+category_id INT PRIMARY KEY AUTO_INCREMENT,
+category_name VARCHAR(25) ,
+parent_category INT,
+FOREIGN KEY (parent_category) REFERENCES category(category_id)
```

Comment:
Can use Add Constraint with a unique name. So that in future if there is need to alter or delete the constraint, it can be done easily.